### ISSUE TITLE
Error with sensor.silla_prism_input_grid_energy

### ISSUE THREAD

**Issue by AndreaDFC** [persuader72/silla-prism-integration#7]
I think there is something broken in the last version (0.5.5).
I have always used "sensor.silla_prism_input_grid_energy" in order to monitor the grid consumption of my house and it always worked well.
But now it gives me a wrong value: for example it indicates that today my home used 0,1775 kWh, really too low.
Is it possible that there is an error in the formula?
Thank you very much for your work!!!

**Comment by persuader72** [comment]
Hi 

The sensor **silla_prism_input_grid_energy** is computed as integral of the sensor **silla_prism_input_grid_power** 
which should measure the instantaneous power read from the silla current sensor. Maybe is the **silla_prism_input_grid_power**  the is reading the wrong values. 

Can you check the reading of **silla_prism_input_grid_power** when your **silla_prism_input_grid_energy** is reporting wrong values?

For example in my case i see something like this:

Power sensor
![image](https://github.com/user-attachments/assets/28a21bbb-3220-4753-b0fa-fe8af91c013c)

Total energy sensor is clearly the integral of the previous measured value.
![image](https://github.com/user-attachments/assets/66043bb8-ef12-4c52-afc7-aa15950adf73)

**Comment by AndreaDFC** [comment]
I think the silla_prism_input_grid_power sensor is working fine. It uses the same values, which I can read in the Diagnostic Page of the Silla Prism.
Maybe I have some error in the integral of the sensor silla_prism_input_grid_power. I thought it could be because I had changed the unit of measurement from Wh to kWh: but I tried to come back to default values and the error was still there.
I would like to point out that the measurement worked well until the previous version.

Power sensor
![image](https://github.com/user-attachments/assets/c601b464-35d8-4d09-8127-b6e5a9d5d17e)

Diagnostic Page from Silla Prism
![image](https://github.com/user-attachments/assets/9ace4600-c396-4996-8ae9-909bcd199a87)

Total energy sensor
![image](https://github.com/user-attachments/assets/421ae77a-bcb2-4899-803f-7a261dd17c56)

**Comment by persuader72** [comment]
You are right I found the bug in the sensor code. The release 0.5.6 solve the problem.

I took the opportunity to change the unit of measurement of the sensor form Wh to kWh. So maybe after update you have to reset the statistics.

Thanks

**Comment by AndreaDFC** [comment]
Now I have the 0.5.6 release, but the problem is still the same. I will update here if I find the reason.
